src/core/src/op/random_uniform.cpp

```diff
@@ -119,12 +119,16 @@ bool op::v8::RandomUniform::visit_attributes(AttributeVisitor& visitor) {
 shared_ptr<Node> op::v8::RandomUniform::clone_with_new_inputs(const OutputVector& new_args) const {
     NGRAPH_OP_SCOPE(v8_RandomUniform_clone_with_new_inputs);
     check_new_args_count(this, new_args);
-    return make_shared<v8::RandomUniform>(new_args[0],
+    auto a = make_shared<v8::RandomUniform>(new_args[0],
                                           new_args[1],
                                           new_args[2],
                                           m_output_type,
                                           m_global_seed,
                                           m_op_seed);
+    // Copy RandomUniform state
+    std::lock_guard<std::mutex> guard(a->m_state_mutex);                                      
```

Comment:
Looks like you don't need this mutex.

Reply:
Thanks, done!

Reply:
Test to be added soon

Reply:
Tests should be added in scope of CVS-69723

Reply:
Added unit test for `clone_with_new_inputs()`